File: AndroidProject/HelloWorld/src/main/java/com/example/helloworld/TestCodelet.java
package com.example.helloworld;

import junit.framework.Test;

import br.unicamp.cst.core.entities.Codelet;

public class TestCodelet extends Codelet {
    private int emotion;
    private float emotionModifier;
    private HelloWorld plugin;

    public TestCodelet(String name, HelloWorld plugin) {
        setName(name);

        this.plugin = plugin;

        emotion = 0;
        emotionModifier = 1;
    }

    @Override
    public void accessMemoryObjects() {

    }

    @Override
    public void calculateActivation() {

    }

    @Override
    public void proc() {
        emotion += emotionModifier;
        if (emotion >= 100) {
            emotionModifier = -1;
        } else if (emotion <= 0) {
            emotionModifier = 1;
        }

        plugin.sendSignal(String.valueOf(emotion));
    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/testcst/TestCodelet.kt
package com.example.game.testcst

import br.unicamp.cst.core.entities.Codelet

class TestCodelet(name: String?) : Codelet() {
    var emotion = 0
        private set
    private var emotionModifier: Float

    override fun accessMemoryObjects() {}

    override fun calculateActivation() {}

    override fun proc() {
        emotion += emotionModifier.toInt()
        if (emotion >= 100) {
            emotionModifier = -1f
        } else if (emotion <= 0) {
            emotionModifier = 1f
        }
    }

    init {
        setName(name)
        emotion = 0
        emotionModifier = 1f
    }
}